domain/modelagent/service/service.go

```diff
@@ -631,8 +646,11 @@ func (s *Service) UpgradeModelTargetAgentVersion(
 	ctx, span := trace.Start(ctx, trace.NameFromFunc())
 	defer span.End()
 
-	recommendedVersion := jujuversion.Current
-	err := s.UpgradeModelTargetAgentVersionTo(ctx, recommendedVersion)
+	recommendedVersion, err := s.getHighestAgentVersion(ctx)
```

Comment:
can we rename this to 'getRecommendedVersion' ?

Reply:
highest is an implementation detail that the caller doesn't need to worry about.